pynumdiff/utils/utility.py

```diff
@@ -109,22 +115,30 @@ def integrate_dxdt_hat(dxdt_hat, _t):
             else cumulative_trapezoid(dxdt_hat, x=_t, initial=0)
 
 
-# Optimization routine to estimate the integration constant.
-def estimate_integration_constant(x, x_hat):
+def estimate_integration_constant(x, x_hat, M=6):
     """Integration leaves an unknown integration constant. This function finds a best fit integration
     constant given x and x_hat (the integral of dxdt_hat) by optimizing :math:`\\min_c ||x - \\hat{x} + c||_2`.
 
     :param np.array[float] x: timeseries of measurements
-    :param np.array[float] x_hat: smoothed estiamte of x, for the purpose of this function this should have been determined
-        by integrate_dxdt_hat
+    :param np.array[float] x_hat: smoothed estimate of x
+    :param float M: robustifies constant estimation using Huber loss. The default is intended to capture the idea
+        of "six sigma": Assuming Gaussian inliers and M in units of standard deviation, the portion of inliers
+        beyond the Huber loss' transition is only about 1.97e-9. M here is in units of scaled mean absolute deviation,
+        so scatter can be calculated and used to normalize data without being thrown off by outliers.
 
     :return: **integration constant** (float) -- initial condition that best aligns x_hat with x
     """
-    return minimize(lambda x0, x, xhat: np.linalg.norm(x - (x_hat+x0)), # fn to minimize in 1st argument
```

Comment:
We were formerly solving the least squares problem here, but the solution to the L2 problem is just the mean of x - x_hat, so now I'm just doing that in the M=infinity case. However, having the minimization problem set up was helpful for seeing how to stick the Huber in there.